image/Dockerfile

```diff
@@ -0,0 +1,64 @@
+# Common builder
+FROM golang:1.10-alpine3.7 as builder
+
+#COPY hack/dockerfile/binaries-commits /
+
+COPY hack/dockerfile/install/tini.installer /
+COPY hack/dockerfile/install/proxy.installer /
+RUN apk --update add bash btrfs-progs-dev gcc libc-dev linux-headers \
+    git cmake make ca-certificates libltdl libtool libgcc && \
+    grep "_COMMIT=" /*.installer  |cut -f2- -d: > /binaries-commits
+
+# dockerd
+FROM builder as dockerd-builder
+WORKDIR /go/src/github.com/docker/docker
+COPY . /go/src/github.com/docker/docker
+ARG VERSION
+ENV VERSION ${VERSION}
+ARG DOCKER_GITCOMMIT
+ENV DOCKER_GITCOMMIT ${DOCKER_GITCOMMIT}
+# TODO The way we set the version could easily be simplified not to depend on hack/...
+RUN bash ./hack/make/.go-autogen
+RUN go build -o /sbin/dockerd \
+    -tags 'autogen netgo static_build selinux journald' \
+    -installsuffix netgo -a -buildmode=pie -ldflags '-w -extldflags "-static" ' \
+    github.com/docker/docker/cmd/dockerd
+
+# docker-proxy
+# TODO if libnetwork folds into the docker tree this can be combined above
+FROM builder as proxy-builder
+RUN git clone https://github.com/docker/libnetwork.git /go/src/github.com/docker/libnetwork
```

Comment:
In a future iteration will we need to have a way to specify a sha to pin down the version of libnetwork that we are using?

Reply:
It's set in the installer script and there's a checkout on line 33